src/index.ts

```diff
@@ -156,11 +160,22 @@ export function createProxyResolver(params: ProxyAgentParams) {
 
 		const { secureEndpoint } = opts as any;
 		const defaultPort = secureEndpoint ? 443 : 80;
-		if (typeof hostname === 'string' && envNoProxy(hostname, String(parsedUrl.port || defaultPort))) {
-			envNoProxyCount++;
-			callback('DIRECT');
-			log.debug('ProxyResolver#resolveProxy envNoProxy', url, 'DIRECT', stackText);
-			return;
+
+		// if there are any config entries present then env variables are ignored
+		if (params.getNoProxyConfig && params.getNoProxyConfig().length) {
```

Comment:
This is already called above. Could you extract it to a variable, so we call it only once?

Reply:
Thanks for the comment. I have extracted it to a new variable `noProxyConfig`